core/src/rpc/rpc_context.rs

```diff
@@ -486,9 +487,33 @@ pub trait RpcContext {
 	// Methods for relating
 	// ------------------------------
 
-	async fn relate(&self, _params: Array) -> Result<impl Into<Data>, RpcError> {
-		let out: Result<Value, RpcError> = Err(RpcError::MethodNotFound);
-		out
+	async fn relate(&self, params: Array) -> Result<impl Into<Data>, RpcError> {
+		let Ok((from, kind, to, content)) = params.needs_three_or_four() else {
+			return Err(RpcError::InvalidParams);
+		};
+
+		let kind = kind.could_be_table();
+
+		// Specify the SQL query string
+		let sql = if content.is_none_or_null() {
+			"RELATE $from->$kind->$to"
+		} else {
+			"RELATE $from->$kind->$to CONTENT $content"
+		};
+
+		// Specify the query parameters
+		let var = Some(map! {
+			String::from("from") => from,
+			String::from("kind") => kind,
+			String::from("to") => to,
+			String::from("content") => content,
+			=> &self.vars()
```

Comment:
These are non-standard and currently not variables which we supply in any other context. Do we need them at all?

Reply:
from, kind, and content are the names of the ast fields so I kept them I use to instead of with because I find with very confusing. I originally generated the ast directly but Tobie asked me to switch to this method

Reply:
Oh sorry, I misunderstood where these variables are applied. Ignore me! 😅